pkg/controller/subscription/subscription_controller.go

```diff
@@ -0,0 +1,93 @@
+* business logic.  Delete these comments after modifying this file.*
+ */
+
+// Add creates a new Subscription Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager, products []string) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileSubscription{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("subscription-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource Subscription
+	err = c.Watch(&source.Kind{Type: &v1alpha1.Subscription{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// blank assignment to verify that ReconcileSubscription implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileSubscription{}
+
+// ReconcileSubscription reconciles a Subscription object
+type ReconcileSubscription struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a Subscription object and makes changes based on the state read
+// and what is in the Subscription.Spec
+// TODO(user): Modify this Reconcile function to implement your Controller logic.  This example creates
+// a Pod as an example
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileSubscription) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling Subscription")
+
+	// Fetch the Subscription instance
+	instance := &v1alpha1.Subscription{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	if instance.Spec.InstallPlanApproval != v1alpha1.ApprovalManual {
+		instance.Spec.InstallPlanApproval = v1alpha1.ApprovalManual
+		r.client.Update(context.TODO(), instance)
```

Comment:
Error handling is missing here. Once added we notice that additional permissions need to be added to operators role. Error:
```
"error":"subscriptions.operators.coreos.com \"integreatly\" is forbidden: 
User \"system:serviceaccount:001-integreatly:integreatly-operator\" 
cannot update resource \"subscriptions\" in API group \"operators.coreos.com\" 
```

Reply:
I have made a new manifest that adds this permission, but I intended to open that PR once this one was merged. I imagined this was being verified by running the operator locally.

Reply:
@philbrookes But there are also cluster roles in deploy/ folder, but I noticed they are slightly out of date already, so we can fix that in one go later...
What do you think of error handling here, is it worth adding?